projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -188,6 +176,53 @@ public void updateDeployment(
     }
   }
 
+  private boolean shouldUpdateDeployment(
+      DataJob dataJob,
+      DesiredDataJobDeployment desiredJobDeployment,
+      ActualDataJobDeployment actualJobDeployment,
+      boolean isJobDeploymentPresentInKubernetes)
+      throws ApiException {
+    if (desiredJobDeployment == null) {
+      log.warn(
+          "Skipping the data job [job_name={}] deployment due to the missing deployment data",
+          dataJob.getName());
+      return false;
+    }
+
+    if (Boolean.FALSE.equals(desiredJobDeployment.getEnabled())) {
+      suspendDeployment(actualJobDeployment, isJobDeploymentPresentInKubernetes);
+      log.trace(
+          "Skipping the data job [job_name={}] deployment due to the suspended deployment",
+          dataJob.getName());
+      return false;
+    }
+
+    if (DeploymentStatus.USER_ERROR.equals(desiredJobDeployment.getStatus())
```

Comment:
Out of curiosity, if a deployment enters an error state, how can it get out of this state, given that it looks like it will always be skipped?

Reply:
If you redeploy the data job through the API, the status will be reset.

Reply:
Got it, thanks